lib/ansible/modules/web_infrastructure/ansible_tower/tower_role.py

```diff
@@ -0,0 +1,226 @@
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+    tower_host:
+      description:
+        - URL to your Tower instance.
+      required: False
+      default: null
+    tower_username:
+        description:
+          - Username for your Tower instance.
+        required: False
+        default: null
+    tower_password:
+        description:
+          - Password for your Tower instance.
+        required: False
+        default: null
+    tower_verify_ssl:
+        description:
+          - Dis/allow insecure connections to Tower. If C(no), SSL certificates will not be validated.
+            This should only be used on personally controlled sites using self-signed certificates.
+        required: False
+        default: True
+    tower_config_file:
+      description:
+        - Path to the Tower config file. See notes.
+      required: False
+      default: null
+
+
+requirements:
+  - "python >= 2.6"
+  - "ansible-tower-cli >= 3.0.3"
+
+notes:
+  - If no I(config_file) is provided we will attempt to use the tower-cli library
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+- name: Add jdoe to the member role of My Team
+  tower_role:
+    user: jdoe
+    target_team: "My Team"
+    role: member
+    state: present
+    tower_config_file: "~/tower_cli.cfg"
+'''
+
+try:
+    import tower_cli
+    import tower_cli.utils.exceptions as exc
+
+    from tower_cli.conf import settings
+    from ansible.module_utils.ansible_tower import tower_auth_config, tower_check_mode
+
+    HAS_TOWER_CLI = True
+except ImportError:
+    HAS_TOWER_CLI = False
+
+
+def update_resources(module, p):
+    '''update_resources attempts to fetch any of the resources given
+    by name using their unique field (identity)
+    '''
+    params = p.copy()
+    identity_map = {
+        'user':'username',
+        'team':'name',
+        'target_team':'name',
+        'inventory':'name',
+        'job_template':'name',
+        'credential':'name',
+        'organization':'name',
+        'project':'name',
+    }
+    for k,v in identity_map.items():
+        try:
+            if params[k]:
+                key = 'team' if k == 'target_team' else k
+                result = tower_cli.get_resource(key).get(**{v:params[k]})
+                params[k] = result['id']
+        except (exc.NotFound) as excinfo:
+            module.fail_json(msg='Failed to update role, {0} not found: {1}'.format(k, excinfo), changed=False)
+    return params
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            user = dict(),
+            team = dict(),
+            role = dict(choices=["admin", "read", "member", "owner", "execute", "adhoc", "update", "use", "auditor"]),
```

Comment:
I don't think owner should be an option. I'll check tower-cli to make sure we have this right.